src/core/radio/radio.hpp

```diff
@@ -741,15 +800,39 @@ inline void Radio::SetPromiscuous(bool aEnable) { otPlatRadioSetPromiscuous(GetI
 
 inline otRadioState Radio::GetState(void) { return otPlatRadioGetState(GetInstancePtr()); }
 
-inline Error Radio::Enable(void) { return otPlatRadioEnable(GetInstancePtr()); }
+inline Error Radio::Enable(void)
+{
+#if OPENTHREAD_CONFIG_RADIO_TIME_STATS_ENABLE && (OPENTHREAD_FTD || OPENTHREAD_MTD)
+    mRadioStatistics.ChangeStatus(RadioStatistics::kSleep);
```

Comment:
Suggestion: Can use `RecordStateChange()` maybe? Makes it clear `RadioStatistics` is just recording the changes and not changing radio state?